src/components/RequireAuth.js

```diff
@@ -297,7 +288,7 @@ const RequireAuth = (props) => {
           >
             <MenuIcon />
           </IconButton>
-          <Button className={classes.appName} onClick={(e) => (window.location.href = "/front")}>
+          <Button className={classes.appName} onClick={(e) => history.push("/")}>
```

Comment:
Is it necessary for this code to work? I think it would be better to separate such fixes.

Reply:
Indeed, it is necessary. In its absence, clicking on the openIMIS logo leads to a complete page refresh, which resets the toggle button to its default state (true). However, with this implementation, while it does redirect to the main page, it avoids an unnecessary reload.